python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/dlt/resource.py

```diff
@@ -67,13 +70,19 @@ def _recursive_cast(value: Any):
         return {k: _recursive_cast(v) for k, v in mapping.items()}
 
     def extract_resource_metadata(
-        self, resource: DltResource, load_info: LoadInfo
+        self,
+        context: Union[OpExecutionContext, AssetExecutionContext],
+        resource: DltResource,
+        load_info: LoadInfo,
+        dlt_pipeline: Pipeline,
```

Comment:
Typically I would recommend making new attributes optional, in case others have overloaded this method.

But, because this is experimental, and not typically something we recommend overloading, we may be good? What do you think?

Reply:
Oh, that's an interesting point. I had never considered overloading a resource since I can't recall seeing an example of that in the docs. 

Personally, I am fine leaving them non-optional to not make the method longer or more clunky (I have at least one more PR that will already make it a bit longer 😅). I also think it being experimental means: "Breaking changes might happen until the API stabilizes, so overload it at your own risk."

But I don't have a very strong opinion either way, so I can definitely make it optional if you think that's better!

Reply:
I agree that the conditional logic would increase quite a bit if made optional, let's leave them!